Amazon.QLDB.Driver.Tests/TestingUtilities.cs

```diff
@@ -125,8 +125,8 @@ public IEnumerable<object[]> GetData(MethodInfo methodInfo)
                     typeof(RetriableException), typeof(InvalidSessionException),
                     Times.Never()},
                 new object[] { new QldbTransactionException(string.Empty, true, new BadRequestException("Bad request")),
-                    typeof(QldbTransactionException), typeof(BadRequestException),
-                    Times.Never()},
+                    typeof(QldbTransactionException), typeof(QldbTransactionException),
```

Comment:
This change is because the old logic would immediately pass a QldbTransactionException through the session, and then it would be unwrapped somewhere later up.

We document QldbTransactionException as something that "is for internal use only", but it was ultimately exposed by the previous implementer. Realistically the user will never receive this exception back from the driver.

The new logic only passes TransactionAbortedException (a child class of QldbTransactionException) directly through (which we throw on Abort). 

This change makes it so that if someone were to throw QldbTransactionException in their lambda for whatever reason, the retry logic would be robust enough to handle retrying on it properly. Basically, if they throw a QldbTransactionException, we wrap that in our own QldbTransactionException with the real context of whether or not it's retriable/session is alive, consume that to handle retries, and then unwrap their original exception and throw it back to them.